.github/workflows/ci.yml

```diff
@@ -0,0 +1,79 @@
+name: CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  rustfmt:
+    name: Check formatting
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+
+      - name: Install rust beta toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: beta
+          profile: minimal
+          override: true
+          components: rustfmt
+
+      - name: Check formatting of all crates
+        run: cargo fmt --all -- --check
+
+  book:
+    name: Build and upload book
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/master'
```

Comment:
FWIW moving this if to the "deploy book" step makes more sense - you likely do want to build things on PRs, just not deploy.

Reply:
That's true